### PR TITLE
Add nonstandard_style alias for bad_style.

### LINKED ISSUE #41646: Add alias 'nonstandard_style' for 'bad_style'
@ashleygwilliams mentioned this in her talk today; `#[allow(bad_style)]` is a rather rebuke-y shame-y annotation on code.

While we do want people to use proper style, it's a useful tool whilst learning to be able to turn off all the _extra_ friction that you deal with that you don't have to. Let's be nicer about it; could we rename this to `nonstandard_style` and keep `bad_style` as a backcompat alias?

Unsure if this requires an RFC.

### PR DESCRIPTION
**Pull request by withoutboats**

### CODE REVIEW

**Comment by Manishearth** [comment]
This isn't enough, right now the bad_style lint group will still be shown in the `rustc -W help` output.

What needs to be done is to rename the lint group and add a silent alias for backwards compatibility.

(also the interaction of overlapping lint groups is kinda iffy)

r=me for landing this as is though.

**Comment by withoutboats** [comment]
Changed the comment to not close the tracking issue, this can be a good first step.

**Comment by Manishearth** [comment]
@bors r+

A simple slightly-hacky way to do this would be to simply treat it as a different name when parsing lint group names.

File: src/test/ui/lint/lint-group-nonstandard-style.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+#[allow(bad_style)]
+mod test {
+    fn CamelCase() {}
+
+    #[forbid(bad_style)]
+    mod bad {
+        fn CamelCase() {} //~ ERROR should have a snake
+
+        static bad: isize = 1; //~ ERROR should have an upper
```
Comment (sgrif) [inline src/test/ui/lint/lint-group-nonstandard-style.rs]: This is such a downer.

**Comment by mark-i-m** [comment]
Lol, why? I'm not sure I understand the motivation for this...

**Comment by steveklabnik** [comment]
@mark-i-m https://github.com/rust-lang/rust/issues/41646

**Comment by mark-i-m** [comment]
Oh, it seems I missed quite an epic thread. It's good to know decisions like that are not made lightly, even though I don't fully agree with the change :)

**Pull request merged**